almalinux-deploy.sh

```diff
@@ -438,9 +524,39 @@ restore_java_links() {
         fi
     done < "${javaBackup}"
     popd
+    save_status_of_stage "restore_java_links"
 }
 
+
+add_efi_boot_record() {
+    if get_status_of_stage "add_efi_boot_record"; then
+        return 0
+    fi
+    if [[ ! -d /sys/firmware/efi ]]; then
+        return
+    fi
+    efibootmgr -c -L "AlmaLinux" -l "\EFI\almalinux\shimx64.efi"
+    report_step_done "The new EFI boot record for AlmaLinux is added"
+    save_status_of_stage "add_efi_boot_record"
+}
+
+
+reinstall_secure_boot_packages() {
+    if get_status_of_stage "reinstall_secure_boot_packages"; then
+        return 0
+    fi
+    for pkg in $(rpm -qa | grep -E 'shim|fwupd|grub2|kernel'); do
```

Comment:
Dangerous. There are only 8.4 kernels available in 8.4 repos.
User will definitely have old kernels from 8.3 installed that we can't reinstall.
We should try to reinstall kernels that we don't have.
Maybe only the latest one?